### PR TITLE
[Internal] Improve Rakefile scripts #trivial

### LINKED ISSUE #743: Homebrew release tweaks
There are some small things we can improve in our `rake release:homebrew` step:
- Check if the upstream `SwiftGen` is available, and if not, add it (pointing to https://github.com/SwiftGen/homebrew-core)
- Check out `master` branch again after the release branch has been pushed

### PR DESCRIPTION
**Pull request by AliSoftware**
* [x] I've started my branch from the `develop` branch (gitflow)
  * [x] And I've selected `develop` as the "base" branch for this Pull Request I'm about to create
* [x] I've added an entry in the `CHANGELOG.md` file to explain my changes and credit myself  
      _(or added `#trivial` to the PR title if I think it doesn't warrant a mention in the CHANGELOG)_
  * [ ] Add the entry in the appropriate section (Bug Fix / New Feature / Breaking Change / Internal Change)
  * [ ] Add a period and 2 spaces at the end of your short entry description
  * [ ] Add links to your GitHub profile to credit yourself and to the related PR and issue after your description

---

This PR:
 - Updates `rubocop` and fixes style warnings/violations triggered by rubocop – commits 1 thru 5
 - Fixes #743 (homebrew fork + switch back to master) – 955f71ebcc2b7d590f92a2dbe07b8ac22d832d24

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should make sure that <code>Podfile.lock</code> and <code>Package.resolved</code> are changed in sync</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Once you fix those tiny nitpickings above, we should be good to go! 🙌

ℹ️ _I will update this comment as you add new commits_

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Podfile [hunk 1]
```diff
@@ -34,11 +36,11 @@ target 'SwiftGenKit' do
   end
 end
 
-post_install do | installer |
+post_install do |_installer|
```
Comment (djbe) [inline Podfile:39]: What's with the underscore here?

File: rakelib/utils.rake [hunk 2]
```diff
@@ -36,12 +38,12 @@ class Utils
 
   def self.podspec_as_json(file)
     file += '.podspec' unless file.include?('.podspec')
-    json, _, _ = Open3.capture3('bundle', 'exec', 'pod', 'ipc', 'spec', file)
+    json, = Open3.capture3('bundle', 'exec', 'pod', 'ipc', 'spec', file)
```
Comment (djbe) [inline rakelib/utils.rake]: This just seems weird to me, but I'm not that familiar with all the ruby syntax conventions, so maybe it's normal? 🤷

File: rakelib/utils.rake [hunk 3]
```diff
@@ -53,9 +55,9 @@ class Utils
 
   def self.pod_trunk_last_version(pod)
     require 'yaml'
-    stdout, _, _ = Open3.capture3('bundle', 'exec', 'pod', 'trunk', 'info', pod)
+    stdout, = Open3.capture3('bundle', 'exec', 'pod', 'trunk', 'info', pod)
     stdout.sub!("\n#{pod}\n", '')
-    last_version_line = YAML.load(stdout).first['Versions'].last
+    last_version_line = YAML.safe_load(stdout).first['Versions'].last
```
Comment (djbe) [inline rakelib/utils.rake:60]: Did rubocop comment on this, or did you notice/improve this?
🤯 if rubocop noticed this.

File: rakelib/release.rake [hunk 4]
```diff
@@ -196,8 +197,11 @@ namespace :release do
       Utils.print_header 'Pushing to Homebrew'
       sh "git add #{formula_file}"
       sh "git commit -m 'swiftgen #{tag}'"
+      # Add remote pointing to our fork, if it doesn't exist yet, then push the new branch to it before opening the PR
+      sh 'git remote add SwiftGen https://github.com/SwiftGen/homebrew-core 2>/dev/null || true'
       sh "git push -u SwiftGen swiftgen-#{tag}"
       sh "open 'https://github.com/Homebrew/homebrew-core/compare/master...SwiftGen:swiftgen-#{tag}?expand=1'"
+      sh 'git checkout master'
```
Comment (AliSoftware) [inline rakelib/release.rake:204]: The actual change for #743

File: Podfile (hunk 1 above)
Comment (AliSoftware) [reply Podfile:39]: The parameter is unused. So rubocop says you should either omit the parameter completely, or name it `_` (like in Swift), or name it whatever but with a leading `_` – to give it a name for readability/code understandability purpose, even if unused.

It's just that `bundle exec rubocop -a` (to auto-correct the violations) apparently chose to go with `_installer` (as opposed to just `_` or removing the parameter completely

File: rakelib/utils.rake (hunk 2 above)
Comment (AliSoftware) [reply rakelib/utils.rake]: Yeah, I might actually think about reverting that. This works and is valid ruby (TIL) but feels odd to me too, so maybe I'll find the corresponding rubocop rule, disable it, and revert that change to not make it too odd.

File: Podfile (hunk 1 above)
Comment (djbe) [reply Podfile:39]: Ah okay that explains it, 👍  for me then.

File: rakelib/utils.rake (hunk 3 above)
Comment (AliSoftware) [reply rakelib/utils.rake:60]: Yep, that's rubocop's doing!

File: rakelib/utils.rake (hunk 2 above)
Comment (djbe) [reply rakelib/utils.rake]: Note that this was changed in a few places.

Comment (AliSoftware) [reply rakelib/utils.rake]: 57c17a41683b94c6dfba4e6171ebc04df4bb63d2

**Review by djbe — APPROVED** [review]

**Pull request merged**